### PR TITLE
listing modules from resistory2

### PR DESCRIPTION
**Pull request by yoshixmk**
# Support
https://deno.land/posts/registry2

## Stop using
Abolished database.json
- 404
  - https://github.com/denoland/deno_website2/blob/master/database.json

### CODE REVIEW

File: .github/workflows/gh-pages.yml [hunk 1]
```diff
@@ -4,8 +4,8 @@ on:
   push:
     branches:
       - master
-  # schedule:
-  #   - cron: '* */6 * * *'
+  schedule:
+    - cron: '0 0,6,12,18 * * *'
```
Comment (jamashita) [inline .github/workflows/gh-pages.yml]: */6

File: mod.ts [hunk 2]
```diff
@@ -98,16 +97,12 @@ if (format === undefined) {
...
 console.debug(green(`Started. format = ${format}`));
 
-const githubDatabaseResource = await resoinseDenoWebsiteGithub();
-const entries: Readonly<Record<string, GithubDatabaseEntry>> =
-  await githubDatabaseResource.json();
+const githubEntries = await ResistoryService.getGithubEntries();
 
 const repositoryPromises: (() => Promise<Repository>)[] = [];
 
-for (const key of Object.keys(entries)) {
```
Comment (jamashita) [inline mod.ts:107]: entries.values()

File: src/services/resistory_service.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        () =>
+          fetchLatestMetaByModuleName(
+            name,
+            versions.get(name)?.latest as string,
+          ).catch((err) => {
+            return null; // TODO Error Object
+          })
+      );
+    return (await concurrentPromise(metaPromises, 100))
+      .filter((meta) => meta !== null) as any as Meta[];
```
Comment (yoshixmk) [inline src/services/resistory_service.ts]: Help me!

File: src/repositories/deno_repository.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
+// Deno Rest API across CDN
+
+import { denoCdn } from "../config.ts";
+import { concurrentPromise } from "../utils/concurrentPromise.ts";
+
+type Version = string;
+type LatestVersion = Version;
```
Comment (robe589) [inline src/repositories/deno_repository.ts]: LatestVersionで置き換えず、Vesionを使えばよいのではないか？

Comment (yoshixmk) [reply src/repositories/deno_repository.ts]: Thx. Fix it!! https://github.com/yoshixmk/deno-x-ranking/pull/5/commits/9d323e30d83f93cf6a38dd9f6231ba88be211abf

File: .github/workflows/gh-pages.yml (hunk 1 above)
Comment (jamashita) [reply .github/workflows/gh-pages.yml]: wao

Comment (jamashita) [reply .github/workflows/gh-pages.yml]: 0 */6

Comment (yoshixmk) [reply .github/workflows/gh-pages.yml]: Thx. Fix it. https://github.com/yoshixmk/deno-x-ranking/pull/5/commits/bf97453a560ad8d3282f2f56cd21f88bdda3b6aa

File: src/services/resistory_service.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  }
+}
+
+// singleton
+export default new ResistoryService() as ResistoryService;
```
Comment (jamashita) [inline src/services/resistory_service.ts]: i prefer named export
```typescript
export const repositoryService = new RepositoryService();
```

File: mod.ts [hunk 6]
```diff
@@ -9,15 +8,15 @@ import {
...
-import { resoinseDenoWebsiteGithub } from "./github.ts";
+import { consoleTable } from "./src/console_table.ts";
+import { generateTsvFile } from "./src/tsv_file_creator.ts";
+import { generateMarkdownFile } from "./src/markdown_file_creator.ts";
+import { sortOrderByDesc as sortOrderByDesc } from "./src/utils/sort.ts";
+import { unique } from "./src/utils/unique.ts";
+import { Repository } from "./src/domains/repository.ts";
+import { concurrentPromise } from "./src/utils/concurrentPromise.ts";
 import { green, Text } from "./deps.ts";
+import ResistoryService from "./src/services/resistory_service.ts";
```
Comment (jamashita) [inline mod.ts]:
```typescript
import { repositoryService } from "./src/services/resistory_service.ts";
```

File: deps.ts [hunk 7]
```diff
@@ -20,3 +19,5 @@ export {
 export {
   PARSE_FAILURE,
 } from "https://deno.land/x/args@2.0.2/symbols.ts";
+import coffee from "https://deno.land/x/coffee@1.0.0/mod.ts";
```
Comment (jamashita) [inline deps.ts]: import goes to the head line

Comment (jamashita) [reply deps.ts]:
```typescrpt
export * from "https://deno.land/x/coffee@1.0.0/mod.ts";
```

Comment (jamashita) [reply deps.ts]:
```typescript
//reexport.ts
export { default } from "d";
```

File: src/services/resistory_service.ts (hunk 5 above)
Comment (yoshixmk) [reply src/services/resistory_service.ts]: fix https://github.com/yoshixmk/deno-x-ranking/pull/5/commits/f8eadd450ae1e0676179454a27ba5e3fcb1e1389

File: mod.ts (hunk 6 above)
Comment (yoshixmk) [reply mod.ts]: fix https://github.com/yoshixmk/deno-x-ranking/pull/5/commits/f8eadd450ae1e0676179454a27ba5e3fcb1e1389

File: deps.ts (hunk 7 above)
Comment (yoshixmk) [reply deps.ts]: we cannot this

File: src/services/resistory_service.ts (hunk 3 above)
Comment (jamashita) [reply src/services/resistory_service.ts]: please use this `array.filter()` method

```typescript
array.filter<S extends T>(callbackfn: (value: T, index: number, array: T[]) => value is S, thisArg?: any): S[];

// this is the normal one.
array.filter(callbackfn: (value: T, index: number, array: T[]) => unknown, thisArg?: any): T[];
```

Comment (yoshixmk) [reply src/services/resistory_service.ts]: Thx. Very good! Fix it!  
https://github.com/yoshixmk/deno-x-ranking/pull/5/commits/7e6140d8f67601608f2da94288cb5e3631002ac7

**Pull request merged**